airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java

```diff
@@ -91,31 +91,48 @@ public void run() {
   @VisibleForTesting
   void submitJob(Job job) {
     final WorkerRun workerRun = workerRunFactory.create(job);
+    // todo (cgardens) - we need to know the attempt id starts before the lifecycle begins otherwise we
+    // can't clean up properly on failures. this hack
+    // assumes that the submitter is always run in a single threaded manner, so we know can figure out
+    // what the next attempt id should be. we should move away from this hack. one approach would be to
+    // switch to uuids.
+    final long attemptId = job.getAttempts().size();
```

Comment:
The way we mix up database id and list index as id here makes me a bit nervous.

Reply:
=> attempt_number